### ISSUE TITLE
SC2120/SC2119 not working when subsequent function call passes arg

### ISSUE THREAD

**Issue by lucianposton** [koalaman/shellcheck#1821]
#### For bugs
- SC2120, SC2119
- My shellcheck version: 0.7.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net
- [ ] and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash


foo() {
    echo "hello ${1}"
}

foo
foo "x"
```

#### Here's what shellcheck currently says:
```
No issues detected!
```

#### Here's what I wanted or expected to see:
```
Line 4:
foo() {
^-- SC2120: foo references arguments, but none are ever passed.
 
Line 8:
foo
^-- SC2119: Use foo "$@" if function's $1 should mean script's $1.
```

**Comment by ugeuder** [comment]
Problem confirmed in version 0.7.1.

Even previous function call with an argument makes the message go away incorrectly.
